server/game/stateWatchers/CardsLeftPlayThisPhaseWatcher.ts

```diff
@@ -0,0 +1,42 @@
+import { Card } from '../core/card/Card';
+import { UnitCard } from '../core/card/CardTypes';
+import { UpgradeCard } from '../core/card/UpgradeCard';
+import { StateWatcherName } from '../core/Constants';
+import Player from '../core/Player';
+import { StateWatcher } from '../core/stateWatcher/StateWatcher';
+import { StateWatcherRegistrar } from '../core/stateWatcher/StateWatcherRegistrar';
+
+export interface CardLeftPlayEntry {
+    card: UnitCard | UpgradeCard;
+    controlledBy: Player;
+}
+
+export type ICardsLeftPlayThisPhase = CardLeftPlayEntry[]
+
+export class CardsLeftPlayThisPhaseWatcher extends StateWatcher<CardLeftPlayEntry[]> {
+    public constructor(registrar: StateWatcherRegistrar, card: Card) {
+        super(StateWatcherName.CardsLeftPlayThisPhase, registrar, card);
+    }
+
+    public override getCurrentValue() {
+        return super.getCurrentValue();
+    }
+
+    public getCardsLeftPlayControlledByPlayer(controller: Player): (UnitCard | UpgradeCard)[] {
```

Comment:
The attack and play watchers both have a `filter` parameter on their corresponding methods to make retrieving results a little more succinct, would be good to have that here as well I think since it would be used in your Boba impl